src/main/java/dev/ecattez/shahmat/board/Board.java

```diff
@@ -0,0 +1,84 @@
+package dev.ecattez.shahmat.board;
+
+import dev.ecattez.shahmat.event.PawnPromoted;
+import dev.ecattez.shahmat.event.PieceCaptured;
+import dev.ecattez.shahmat.event.PiecePositioned;
+import dev.ecattez.shahmat.event.PromotionProposed;
+import dev.ecattez.shahmat.event.PieceMoved;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class Board {
+
+    private final Map<Square, Piece> boardState;
+
+    public Board(Map<Square, Piece> boardState) {
+        this.boardState = boardState;
+    }
+
+    public Board() {
+        this.boardState = new HashMap<>();
+    }
+
+    public Optional<Piece> getPiece(Square location) {
+        return Optional.ofNullable(boardState.get(location));
+    }
+
+    public Piece unsafeGetPiece(Square location) throws NoPieceOnSquare {
```

Comment:
Not sure that the `unsafe` has any meaning here. The intent is already displayed with the checked exception.

Reply:
I could change `Optional getPiece` by `Optional findPiece` to be more concise and refactor `unsafeGetPiece` by `getPiece`.